### DISCUSSION TITLE
Accessing computed properties inside JS ($wire.computedProp)

### DISCUSSION THREAD

**Discussion by smarterdigitalltd** [livewire/livewire#7050] [1. Help]
Hi there

I'm wondering how to access Livewire 3.0 computed properties from within Alpine.

I see from the docs that there is a different syntax for computed props inside PHP (needing to use $this), but I'm wondering if anything needs to be done JS side..?

I get `Cannot call [myComputedProp()] computed property method directly on component: my-component` with the following in my template

```
<div
    x-show="$wire.myComputedProp"
    class="fixed inset-0 bg-gray-500 bg-opacity-25 transition-opacity"
    wire:transition.enter="ease-out duration-300"
    wire:transition.enter-start="opacity-0"
    wire:transition.enter-end="opacity-100"
    wire:transition.leave="ease-in duration-200"
    wire:transition.leave-start="opacity-100"
    wire:transition.leave-end="opacity-0"
></div>
```

Thanks

**Comment by smarterdigitalltd** [comment]
I'm following up as I think my syntax was way off here. I saw a mention of an `@entangle` directive from the Livewire 2.0 docs + I see it's been referenced in the upgrade guide for v3 and it looks like what I need to be using. Nevertheless, I'm still not able to "entangle" a computed prop it seems. I see this error in the JS console: 

`Livewire Entangle Error: Livewire property ['show'] cannot be found on component: ['my-component']`

And here is my template code:

```
<div x-data="{ show: @entangle('show').live }">
    <div x-show="show"></div>
</div>
```

I'm also unable to see a reference to the computed property if I console.log `$wire.__instance`.

Any thoughts/pointers would be appreciated.

**Comment by sektormediaofficial** [comment]
I have the exact same problem, any ideas here? My use case is I want to scroll the window of laravel echo chat, once a new message is received, and my messages come from a computed property `messageList`

**Reply by edalzell** [reply]
maybe use watch?

**Reply by DanielSpravtsev** [reply]
> I have the exact same problem, any ideas here? My use case is I want to scroll the window of laravel echo chat, once a new message is received, and my messages come from a computed property `messageList`

In your case, you can dispatch from php and listen with alpine using x-on:event_name.window

**Comment by paul-adams-985** [comment]
Same here, my code would be beautiful if I could $wire.get('computedProperty'). But I get undefined.

I notice that you must include $this when using blade to display {{ $this->computedProperty }} which you don't need to do for standard properties. Maybe that is a clue.

**Comment by joshhanley** [comment]
Computed properties aren't accessible on the front end, they are only available on the backend.

**Reply by firer777** [reply]
No, JS can change value Computed properties by Livewire.find('componentId').set('computedProperty', 'new Value');

**Reply by firer777** [reply]
I added new #[Locked] to computed property but not work, can change by js

**Reply by A-Moussa0** [reply]
@sinhnv1991 is this still the case?

**Reply by firer777** [reply]
@A-Moussa0 No livewire did fix on new version

**Comment by EPuigdemont** [comment]
My workaround to fill select2 elements with queryable data is to use a renderless method that is called by alpine, and that renderless method simply does: 

`return $this->computedProperty;`

All it does is a simple ajax request to the renderless method as a proxy for the computed property values, works quite fast. This also means that any attributes you would like to pass to a computed property, or if you need to clear it, can be handled by that renderless method as well.

**Reply by RealMrHex** [reply]
can u share any code or sample plz?
